### DISCUSSION TITLE
✨Request for more metadata and functionality in web browser search results

### DISCUSSION THREAD

**Discussion by mastreips** [policratus/pupyl#103] [Ideas]
**Is your feature request related to a problem? Please describe.**
I don't get as much utility from the web search results as I would like. Currently, I only have access to a static image and the associated filename. 

**Describe the solution you'd like**
I would love to see more metadata with the Web Search results, including file location and a URI to the location so I can open the folder where the image is located. It would be nice to access individual images from the search results.  It would be nice to be able to do some image collection management with the web service itself (delete, move, etc.)

**Describe alternatives you've considered**
I have looked at other image similarity solutions on github that do have the functionality I am seeking, such as https://github.com/zazaho/SimImg However, I would like to take advantage of the interesting deep learning (EfficiencyNet and DenseNet) functionality of this tool since it does a very nice of finding interesting pattern associations across image categories.  

**Additional context**
N/A

**Comment by policratus** [comment]
Hey @mastreips, thanks for reporting this! Let's talk about every point.

1. ~**Metadata**: Yep, certainly we'll bring more file metadata to the web interface, also adding links to individual images. It will be on `pupyl` version `0.12.3`, thanks for this idea! Will be merged to upstream after closing issue #104.~
2. ~**Collection management**: There's a discussion (#70) and a issue (#80) which partially refers to management of the database. This is a relevant improvement to the library and soon as possible will be part of the main stream.~
3. **GUI management**: Although an interface like shown on [SimImg](https://github.com/zazaho/SimImg) would be really welcomed to `pupyl`, having this is not the main intention of the library. Hence, collection management on API level definitely will be developed, but bringing this to a GUI is uncertain. The main focus of `pupyl` is to bring the best and state-of-the-art image similarity algorithms and techniques to the library and we hope get more (specialized) developers helping to make an incredible GUI or user interface soon.

**Comment by policratus** [comment]
Hi @mastreips.

The above topic **Metadata** is done and #104 was already closed. The version `0.12.3` is generally available too.

Please enjoy.

**Comment by policratus** [comment]
Hey @mastreips,

The collection management topic was implemented and is part of `pupyl` `0.12.4` (merged by #111 and discussed on #70).
